Kommunicate/Classes/KMConversationListViewController.swift

```diff
@@ -0,0 +1,571 @@
+            var groupId: NSNumber?
+            var contactId: String?
+            var conversationId: NSNumber?
+
+            if components.count > 2 {
+                let groupComponent = Int(components[1])
+                groupId = NSNumber(integerLiteral: groupComponent!)
+            } else if components.count == 2 {
+                let conversationComponent = Int(components[1])
+                conversationId = NSNumber(integerLiteral: conversationComponent!)
+                contactId = components[0]
+            } else {
+                contactId = object
+            }
+
+            let message = ALMessage()
+            message.contactIds = contactId
+            message.groupId = groupId
+            let info = notification.userInfo
+            let alertValue = info?["alertValue"]
+            guard let updateUI = info?["updateUI"] as? Int else { return }
+            if updateUI == Int(APP_STATE_ACTIVE.rawValue), weakSelf.isViewLoaded, weakSelf.view.window != nil {
+                guard let alert = alertValue as? String else { return }
+                let alertComponents = alert.components(separatedBy: ":")
+                if alertComponents.count > 1 {
+                    message.message = alertComponents[1]
+                } else {
+                    message.message = alertComponents.first
+                }
+                weakSelf.viewModel.addMessages(messages: [message])
+            } else if updateUI == Int(APP_STATE_INACTIVE.rawValue) {
+                // Coming from background
+
+                guard contactId != nil || groupId != nil || conversationId != nil else { return }
+                weakSelf.launchChat(contactId: contactId, groupId: groupId, conversationId: conversationId)
+            }
+        })
+
+        NotificationCenter.default.addObserver(forName: NSNotification.Name(rawValue: "reloadTable"), object: nil, queue: nil, using: { [weak self] notification in
+            NSLog("Reloadtable notification received")
+
+            guard let weakSelf = self, let list = notification.object as? [Any] else { return }
+            weakSelf.viewModel.updateMessageList(messages: list)
+        })
+
+        NotificationCenter.default.addObserver(forName: NSNotification.Name(rawValue: "USER_DETAILS_UPDATE_CALL"), object: nil, queue: nil, using: { [weak self] notification in
+            NSLog("update user detail notification received")
+
+            guard let weakSelf = self, let userId = notification.object as? String else { return }
+            print("update user detail")
+            weakSelf.viewModel.updateUserDetail(userId: userId) { (success) in
+                if success {
+                    weakSelf.tableView.reloadData()
+                }
+            }
+        })
+
+        NotificationCenter.default.addObserver(forName: NSNotification.Name(rawValue: "UPDATE_CHANNEL_NAME"), object: nil, queue: nil, using: { [weak self] _ in
```

Comment:
We should use `#selector` instead of closure otherwise observers won't be removed.